javascriptv3/example_code/ssm/scenarios/ssm-basics.js

```diff
@@ -0,0 +1,447 @@
+  }
+);
+
+export const sdkDeleteResources = new ScenarioAction(
+  "sdkDeleteResources",
+  /**
+   * @param {{ }} c
+   */
+  async (c) => {
+    //delete ops item
+    const opsItemResponse = await client.send(
+      new DeleteOpsItemCommand({
+        OpsItemId: c.opsItemId,
+      })
+    );
+    if (opsItemResponse.$metadata.httpStatusCode == 200) {
+      console.log(
+        "The ops item: " + c.opsItemId + " was successfully deleted."
+      );
+    } else {
+      console.log(
+        "There was a problem deleting the ops item: " +
+          c.opsItemId +
+          ". Please delete it manually."
+      );
+    }
+
+    const maintenanceWindowResponse = await client.send(
+      new DeleteMaintenanceWindowCommand({
+        Name: c.maintenanceWindow,
+        WindowId: c.winId,
+      })
+    );
+    if (maintenanceWindowResponse.$metadata.httpStatusCode == 200) {
+      console.log(
+        "The maintenance window: " +
+          c.maintenanceWindow +
+          " was successfully deleted."
+      );
+    } else {
+      console.log(
+        "There was a problem deleting the maintenance window: " +
+          c.opsItemId +
+          ". Please delete it manually."
+      );
+    }
+
+    const docResponse = await client.send(
+      new DeleteDocumentCommand({
+        Name: c.documentName,
+      })
+    );
+    if (docResponse.$metadata.httpStatusCode == 200) {
+      console.log(
+        "The document: " + c.documentName + " was successfully deleted."
+      );
+    } else {
+      console.log(
+        "There was a problem deleting the document: " +
+          c.documentName +
+          ". Please delete it manually."
+      );
+    }
+  },
+  { skipWhen: (/** @type {{}} */ state) => !state.askToDeleteResources }
+);
+
+const goodbye = new ScenarioOutput(
+  "goodbye",
+  "This concludes the Systems Manager Basics scenario for the AWS Javascript SDK v3. Thank you!"
+);
+
+const myScenario = new Scenario("SSM Basics", [
+  greet,
+  pressEnter,
+  createMaintenanceWindow,
+  getMaintenanceWindow,
+  sdkCreateMaintenanceWindow,
+  modifyMaintenanceWindow,
+  pressEnter,
+  sdkModifyMaintenanceWindow,
+  createSystemsManagerActions,
+  getDocumentName,
+  sdkCreateSSMDoc,
+  ec2HelloWorld,
+  confirmEC2HelloWorld,
+  sdkEC2HelloWorld,
+  pressEnter,
+  getCommandTime,
+  sdkGetCommandTime,
+  pressEnter,
+  showCommandTime,
+  createSSMOpsItem,
+  pressEnter,
+  sdkCreateSSMOpsItem,
+  updateOpsItem,
+  pressEnter,
+  sdkUpdateOpsItem,
+  getOpsItemStatus,
+  pressEnter,
+  sdkOpsItemStatus,
+  resolveOpsItem,
+  pressEnter,
+  sdkResolveOpsItem,
+  askToDeleteResources,
+  confirmDeleteChoice,
+  pressEnter,
+  sdkDeleteResources,
+  goodbye,
+]);
+
+export const main = async (stepHandlerOptions) => {
+  await myScenario.run(stepHandlerOptions);
+};
+
+// Invoke main function if this file was run directly.
+if (process.argv[1] === fileURLToPath(import.meta.url)) {
+  main();
```

Comment:
The user needs to be able to pass options here. `parseScenarioArgs` works for that.